sw/host/opentitanlib/src/util/openocd.rs

```diff
@@ -0,0 +1,256 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::borrow::BorrowMut;
+use std::cell::{Cell, RefCell};
+use std::io::Read;
+use std::io::Write;
+use std::net::TcpStream;
+use std::process::{Child, Command, Stdio};
+use std::thread;
+use std::time::Duration;
+
+use anyhow::{bail, Context, Result};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::dif::lc_ctrl::LcCtrlReg;
+use crate::impl_serializable_error;
+use crate::io::jtag::{Jtag, JtagParams, JtagTap};
+
+/// Port to spawn and connect to OpenOCD over.
+const OPENOCD_TCL_PORT: &str = "6666";
+
+/// Represents an OpenOCD server that we can interact with.
+pub struct OpenOcdServer {
+    /// JTAG parameters
```

Comment:
Nit. Maybe finish these comments with a `.` to be consistent with the other comments in the file?